src/storage/txn/process.rs

```diff
@@ -517,6 +522,13 @@ fn process_write_impl<S: Snapshot, L: LockManager>(
     statistics: &mut Statistics,
     pipelined_pessimistic_lock: bool,
 ) -> Result<WriteResult> {
+    let need_calc_lock_hash = || {
```

Comment:
Why not evaluate it eagerly?

Reply:
Some commands don't need it. I'm not sure whether it's worthy...

Reply:
Ah, I see.